daemon/api.go

```diff
@@ -2158,12 +2158,13 @@ func readyToBuy(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 func runSnapctl(c *Command, r *http.Request, user *auth.UserState) Response {
-	var snapctlOptions client.SnapCtlOptions
-	if err := jsonutil.DecodeWithNumber(r.Body, &snapctlOptions); err != nil {
+	var snapctlPostData client.SnapCtlPostData
+
+	if err := jsonutil.DecodeWithNumber(r.Body, &snapctlPostData); err != nil {
```

Comment:
Same question about imposing a cap on stdin data. Main concern is OOM and also huge data blob serialized into hook's context in the state if a buggy caller spams the hook with too much data.